gammapy/analysis/core.py

```diff
@@ -80,90 +63,48 @@ def config(self, value):
         else:
             raise TypeError("config must be dict or AnalysisConfig.")
 
-    def _set_data_store(self):
-        """Set the datastore on the Analysis object."""
-        path = make_path(self.config.observations.datastore)
-        if path.is_file():
-            log.debug(f"Setting datastore from file: {path}")
-            self.datastore = DataStore.from_file(path)
-        elif path.is_dir():
-            log.debug(f"Setting datastore from directory: {path}")
-            self.datastore = DataStore.from_dir(path)
-        else:
-            raise FileNotFoundError(f"Datastore not found: {path}")
-
-    def _make_obs_table_selection(self):
-        """Return list of obs_ids after filtering on datastore observation table."""
-        obs_settings = self.config.observations
-
-        # Reject configs with list of obs_ids and obs_file set at the same time
-        if len(obs_settings.obs_ids) and obs_settings.obs_file is not None:
-            raise ValueError(
-                "Values for both parameters obs_ids and obs_file are not accepted."
-            )
-
-        # First select input list of observations from obs_table
-        if len(obs_settings.obs_ids):
-            selected_obs_table = self.datastore.obs_table.select_obs_id(
-                obs_settings.obs_ids
-            )
-        elif obs_settings.obs_file is not None:
-            path = make_path(obs_settings.obs_file)
-            ids = list(Table.read(path, format="ascii", data_start=0).columns[0])
-            selected_obs_table = self.datastore.obs_table.select_obs_id(ids)
+    def run(self, steps=None, overwrite=None, **kwargs):
```

Comment:
add doctring

Reply:
what does the `overwrite` option do?

Reply:
I had in mind that each AnalysisStep could have a read/write method but for now this is used only in the DataReductionAnalysisStep to read the datasets if exits.